stan/math/rev/fun/multiply.hpp

```diff
@@ -592,6 +592,8 @@ inline auto multiply(const Mat1& m1, const Mat2& m2) {
  */
 template <
     typename RowVec, typename ColVec,
+    require_all_eigen_vt<is_var_or_arithmetic, RowVec, ColVec>* = nullptr,
+    require_any_eigen_vt<is_var, RowVec, ColVec>* = nullptr,
```

Comment:
Why is this require_any_eigen_vt required?

Reply:
Let me rethink this. The logic here is that we want to accept combinations of vars and arithmetic types but at least one of the types has to have an var scalar.